x-pack/plugins/maps/public/routing/routes/maps_app/maps_app.js

```diff
@@ -0,0 +1,508 @@
+      storeUpdates.prevIndexPatternIds = nextIndexPatternIds;
+      await this._updateIndexPatterns(nextIndexPatternIds);
+    }
+
+    if (nextIsSaveDisabled !== isSaveDisabled) {
+      storeUpdates.isSaveDisabled = nextIsSaveDisabled;
+    }
+
+    const nextIsOpenSettingsDisabled = flyoutDisplay !== FLYOUT_STATE.NONE;
+    if (nextIsOpenSettingsDisabled !== isOpenSettingsDisabled) {
+      storeUpdates.isOpenSettingsDisabled = nextIsOpenSettingsDisabled;
+    }
+    if (!_.isEmpty(storeUpdates)) {
+      this.setState(storeUpdates);
+    }
+  };
+
+  _getAppStateFilters = () => {
+    return this._appStateManager.getFilters() || [];
+  };
+
+  _syncAppAndGlobalState = () => {
+    const { query, time, refreshConfig, initialized } = this.state;
+    const { filterManager } = getData().query;
+
+    // appState
+    this._appStateManager.setQueryAndFilters({
+      query: query,
+      filters: filterManager.getAppFilters(),
+    });
+
+    // globalState
+    const refreshInterval = {
+      pause: refreshConfig.isPaused,
+      value: refreshConfig.interval,
+    };
+    updateGlobalState(
+      {
+        time: time,
+        refreshInterval,
+        filters: filterManager.getGlobalFilters(),
+      },
+      !initialized
+    );
+    this.setState({ refreshInterval });
+  };
+
+  _onQueryChange = async ({ filters, query, time, refresh }) => {
+    const { filterManager } = getData().query;
+    const { dispatchSetQuery } = this.props;
+    const newState = {};
+    let newFilters;
+    if (filters) {
+      filterManager.setFilters(filters); // Maps and merges filters
+      newFilters = filterManager.getFilters();
+      newState.filters = newFilters;
+    }
+    if (query) {
+      newState.query = query;
+    }
+    if (time) {
+      newState.time = time;
+    }
+    this.setState(newState, () => {
+      this._syncAppAndGlobalState();
+      dispatchSetQuery(
+        refresh,
+        newFilters || this.state.filters,
+        query || this.state.query,
+        time || this.state.time
+      );
+      updateGlobalState(
+        { ...newState, filters: filterManager.getGlobalFilters() },
+        !this.state.initialized
+      );
+    });
+  };
+
+  _initQueryTimeRefresh(savedMap) {
+    const { setRefreshConfig } = this.props;
+    // TODO: Handle null when converting to TS
+    const globalState = getGlobalState();
+    const mapStateJSON = savedMap ? savedMap.mapStateJSON : undefined;
+    const newState = {
+      query: getInitialQuery({
+        mapStateJSON,
+        appState: this._appStateManager.getAppState(),
+        userQueryLanguage: getUiSettings().get('search:queryLanguage'),
+      }),
+      time: getInitialTimeFilters({
+        mapStateJSON,
+        globalState,
+      }),
+      refreshConfig: getInitialRefreshConfig({
+        mapStateJSON,
+        globalState,
+      }),
+    };
+    this.setState(newState);
+    updateGlobalState(
+      {
+        query: newState.query,
```

Comment:
Query should not be passed to global state. Notice in the URL below how query is in `_g` and `_a`

```
http://localhost:5603/xik/app/maps/map/98946990-b0af-11ea-980d-757daf12e307#?_g=(filters:!(),query:(language:kuery,query:'machine.os.keyword%20:%20%22win%207%22%20'),refreshInterval:(pause:!t,value:0),time:(from:now-7d,to:now))&_a=(filters:!(),query:(language:kuery,query:'machine.os.keyword%20:%20%22win%207%22%20'))
```